pkg/kv/kvserver/replica_application_result.go

```diff
@@ -507,10 +507,19 @@ func (r *Replica) stagePendingTruncationRaftMuLocked(pt pendingTruncation) {
 	// safe to consider the log size delta trusted in this case. Conveniently,
 	// this doesn't need any special casing.
 	isDeltaTrusted := pt.isDeltaTrusted && r.shMu.raftTruncState.Index+1 == pt.expectedFirstIndex
+
 	r.mu.Lock()
 	r.shMu.raftTruncState = pt.RaftTruncatedState
-	r.handleRaftLogDeltaResultRaftMuLockedReplicaMuLocked(pt.logDeltaBytes, isDeltaTrusted)
+	// Ensure the raft log size is not negative since it isn't persisted between
+	// server restarts.
+	// TODO(pav-kv): should we distrust the log size if it goes negative?
```

Comment:
That makes sense to me at least. But (outside of the expectedFirstIndex==0 case where all bets are off) it should not be possible to arrive at a negative number through trusted updates, correct?

Reply:
Yes, we shouldn't see a negative here, post all the cleanups. There are still a couple of possibilities:

1. The sideloaded storage size tracking is faulty and can silently skew the log size: #136416.
2. I don't have full confidence that the leaseholder-evaluated size delta exactly matches what the local replica would have computed. Maybe not today, but I could imagine mixed-version scenarios in which raft log encodings could differ.